docs/monitoring/back_pressure.md

```diff
@@ -34,30 +34,28 @@ If you see a **back pressure warning** (e.g. `High`) for a task, this means that
 Take a simple `Source -> Sink` job as an example. If you see a warning for `Source`, this means that `Sink` is consuming data slower than `Source` is producing. `Sink` is back pressuring the upstream operator `Source`.
 
 
-## Sampling Threads
+## Sampling Tasks
 
-Back pressure monitoring works by repeatedly taking stack trace samples of your running tasks. The JobManager triggers repeated calls to `Thread.getStackTrace()` for the tasks of your job.
+Back pressure monitoring works by repeatedly taking samples of your running tasks. The JobManager triggers repeated calls to `Task.isBackPressured()` for the tasks of your job.
 
 <img src="{{ site.baseurl }}/fig/back_pressure_sampling.png" class="img-responsive">
-<!-- https://docs.google.com/drawings/d/1_YDYGdUwGUck5zeLxJ5Z5jqhpMzqRz70JxKnrrJUltA/edit?usp=sharing -->
+<!-- https://docs.google.com/drawings/d/1O5Az3Qq4fgvnISXuSf-MqBlsLDpPolNB7EQG7A3dcTk/edit?usp=sharing -->
 
-If the samples show that a task Thread is stuck in a certain internal method call (requesting buffers from the network stack), this indicates that there is back pressure for the task.
```

Comment:
Add some introductions of how to judge the back pressure state instead.

E.g. Internally we judge the back pressure based on the availability of output buffers. If there are no available buffers for at-least one output, then it indicates that there is back pressure for the task.